Sources/OrderedCollections/OrderedSet/OrderedSet.swift

```diff
@@ -494,3 +494,26 @@ extension OrderedSet {
     return _elements.remove(at: index)
   }
 }
+
+extension OrderedSet {
+  /// Returns a new ordered set containing the values pairs of the ordered set
+  /// that satisfy the given predicate.
+  ///
+  /// - Parameter isIncluded: A closure that takes a value as its
+  ///   argument and returns a Boolean value indicating whether the value
+  ///   should be included in the returned dictionary.
+  ///
+  /// - Returns: An ordered set of the values that `isIncluded` allows.
+  ///
+  /// - Complexity: O(`count`)
+  @inlinable
+  public func filter(
+          _ isIncluded: (Element) throws -> Bool
+  ) rethrows -> Self {
```

Comment:
This implementation can be improved by using a temporary bitset, as in [`Set.filter`](https://github.com/apple/swift/blob/main/stdlib/public/core/NativeSet.swift#L722-L736) -- I expect this will boost performance by 2x or more, by eliminating the need to repeatedly resize the hash table.

(This does not need to be implemented before this lands.)

Reply:
Hm, I poked a little bit at it but we end up having to pull in quite a bunch of new uncheckedElement, _bucketCounts etc I'm not super sure if I'm implementing them right -- so leaving this as future work I guess. Happy to make a ticket for it after this lands.

Reply:
Oh, the stdlib code wouldn't directly translate -- that's just illustrating the concept. We have a separate bitset implementation within this code base, so we'd need to adapt the algorithm to use that instead. ([OrderedSet.symmetricDifference](https://github.com/apple/swift-collections/blob/2ce28a9eaac3a03c7403b09aae10fea3273de193/Sources/OrderedCollections/OrderedSet/OrderedSet%2BPartial%20SetAlgebra%2BOperations.swift#L290-L314) is an example.)

I'm happy to do this as a followup.

Reply:
Followup: https://github.com/apple/swift-collections/pull/163